syntax = "proto3";

service Reservations {
  rpc makeReservation (Reservation) returns (ReservationResponse);
  rpc checkReservation (ReservationId) returns (ReservationResponse);
  rpc cancelReservation (ReservationId) returns (CancelResponse);
  rpc listReservations (ListRequest) returns (ReservationsList);
}

enum Result {
    OK = 0;
    ERROR = 1;
}

message ReservationId {
  string date = 1;
}

message Reservation {
  string date = 1;
  int32 time = 2;
  int32 number_of_diners = 3;
  string client_name = 4;
  string contact_phone = 5;
}

message ReservationResponse {
  Result result = 1;
  string details = 2;
  Reservation reservation = 3;
}

message CancelResponse {
  Result result = 1;
  string details = 2;
}

message ListRequest {}

message ReservationsList {
  repeated Reservation reservations = 1;
}
